version: "3"

x-container-common: &container-common
  restart: ${CONTAINER_RESTART-on-failure}
  networks:
    - deps-pg

services:
  pg:
    <<: *container-common
    image: postgres:13.4-alpine
    container_name: pg
    environment:
      DATABASE_USER: ${DATABASE_USER:-postgres}
      DATABASE_PASSWORD: ${DATABASE_PASSWORD:-postgres}
    volumes:
      - pg-data:/var/lib/postgresql/data
    ports:
      - 5432:5432
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres" ]
      interval: 1s
      timeout: 1s
      retries: 5

  pgadmin:
    <<: *container-common
    image: dpage/pgadmin4
    container_name: pg-admin
    environment:
      PGADMIN_DEFAULT_EMAIL: ${PGADMIN_DEFAULT_EMAIL-pgadmin4@pgadmin.org}
      PGADMIN_DEFAULT_PASSWORD: ${PGADMIN_DEFAULT_PASSWORD-admin}
      PGADMIN_CONFIG_SERVER_MODE: 'False'
    volumes:
      - pgadmin-data:/var/lib/pgadmin/data
    ports:
      - ${PGADMIN_PORT-5050}:80

networks:
  deps-pg:
    name: deps-pg-dev
    driver: bridge

volumes:
  pg-data:
    driver: local
  pgadmin-data:
    driver: local